Description: Add disable opposites to the security-related flags
Author: Stephen Kitt <skitt@debian.org>

This patch adds "no-" variants to disable the various security flags:
"no-dynamicbase", "no-nxcompat", "no-high-entropy-va". These are
equivalent to the corresponding "disable-" flags in binutils 2.36, and
are only preserved for backwards compatibility with previous versions
of the Debian packages.

--- a/upstream/ld/emultempl/pe.em
+++ b/upstream/ld/emultempl/pe.em
@@ -286,8 +286,11 @@
   OPTION_DISABLE_LONG_SECTION_NAMES,
 /* DLLCharacteristics flags.  */
   OPTION_DYNAMIC_BASE,
+  OPTION_NO_DYNAMIC_BASE,
   OPTION_FORCE_INTEGRITY,
   OPTION_NX_COMPAT,
+  OPTION_NO_NX_COMPAT,
+  OPTION_ISOLATION,
   OPTION_NO_ISOLATION,
   OPTION_NO_SEH,
   OPTION_NO_BIND,
@@ -384,10 +387,12 @@
     {"disable-long-section-names", no_argument, NULL, OPTION_DISABLE_LONG_SECTION_NAMES},
     {"dynamicbase",no_argument, NULL, OPTION_DYNAMIC_BASE},
     {"disable-dynamicbase",no_argument, NULL, OPTION_DISABLE_DYNAMIC_BASE},
+    {"no-dynamicbase",no_argument, NULL, OPTION_DISABLE_DYNAMIC_BASE},
     {"forceinteg", no_argument, NULL, OPTION_FORCE_INTEGRITY},
     {"disable-forceinteg", no_argument, NULL, OPTION_DISABLE_FORCE_INTEGRITY},
     {"nxcompat", no_argument, NULL, OPTION_NX_COMPAT},
     {"disable-nxcompat", no_argument, NULL, OPTION_DISABLE_NX_COMPAT},
+    {"no-nxcompat", no_argument, NULL, OPTION_DISABLE_NX_COMPAT},
     {"no-isolation", no_argument, NULL, OPTION_NO_ISOLATION},
     {"disable-no-isolation", no_argument, NULL, OPTION_DISABLE_NO_ISOLATION},
     {"no-seh", no_argument, NULL, OPTION_NO_SEH},
--- a/upstream/ld/emultempl/pep.em
+++ b/upstream/ld/emultempl/pep.em
@@ -399,9 +399,12 @@
     {"enable-reloc-section", no_argument, NULL, OPTION_ENABLE_RELOC_SECTION},
     {"disable-reloc-section", no_argument, NULL, OPTION_DISABLE_RELOC_SECTION},
     {"disable-high-entropy-va", no_argument, NULL, OPTION_DISABLE_HIGH_ENTROPY_VA},
+    {"no-high-entropy-va", no_argument, NULL, OPTION_DISABLE_HIGH_ENTROPY_VA},
     {"disable-dynamicbase",no_argument, NULL, OPTION_DISABLE_DYNAMIC_BASE},
+    {"no-dynamicbase",no_argument, NULL, OPTION_DISABLE_DYNAMIC_BASE},
     {"disable-forceinteg", no_argument, NULL, OPTION_DISABLE_FORCE_INTEGRITY},
     {"disable-nxcompat", no_argument, NULL, OPTION_DISABLE_NX_COMPAT},
+    {"no-nxcompat", no_argument, NULL, OPTION_DISABLE_NX_COMPAT},
     {"disable-no-isolation", no_argument, NULL, OPTION_DISABLE_NO_ISOLATION},
     {"disable-no-seh", no_argument, NULL, OPTION_DISABLE_NO_SEH},
     {"disable-no-bind", no_argument, NULL, OPTION_DISABLE_NO_BIND},
